demo/admin/src/formBuilder/utils/createFieldBlock.ts

```diff
@@ -0,0 +1,49 @@
+import { BlockCategory, BlockInterface, createBlockSkeleton } from "@comet/blocks-admin";
+
+import { RichTextBlock } from "../blocks/common/RichTextBlock";
+
+type RequiredSettings = "name" | "displayName" | "defaultValues" | "AdminComponent";
+
+type Settings = Pick<BlockInterface, RequiredSettings> & Partial<Omit<BlockInterface, RequiredSettings>>;
+
+export const createFieldBlock = (settings: Settings): BlockInterface => {
```

Comment:
This seems like a hasty abstraction to me. Does it offer that match benefit? It's also rather loosely typed...

Reply:
I mainly wanted to abstract away `previewContent`, `isValid`, `input2State`, `state2Output`, `output2State` and `createPreviewState`. 

Maybe something like `createBlockSkeleton` would be better, e.g. `...createFormBuilderBlockSkeleton()`.
Would you prefer that?